packages/flutter/lib/src/widgets/form.dart

```diff
@@ -141,6 +141,15 @@ class Form extends StatefulWidget {
   /// {@macro flutter.widgets.form.autovalidateMode}
   final AutovalidateMode autovalidateMode;
 
+  /// Used to enable/disable form fields auto validation and update their error
+  /// text.
+  @Deprecated(
+    'Use autoValidateMode parameter which provide more specific '
+    'behaviour related to auto validation. '
+    'This feature was deprecated after v1.19.0.'
+  )
+  final bool autovalidate;
```

Comment:
Is there a reason to put this as a actual value instead of a getter?

Consider the behavior:
- Form created using `autovalidateMode` to `AutovalidateMode.always` in constructor
    - Access `autovalidate` will be `false`, while `autovalidateMode` will be `AutovalidateMode.always`
- Form created setting `autovalidate` to `true` in constructor
    - Access `autovalidate` will be `true`, while `autovalidateMode` will be `AutovalidateMode.always`

Putting this as a getter referencing `autovalidateMode` means there's only a single property storing state, so it can't be inconsistent.
```suggestion
  bool get autovalidate => autovalidateMode != AutovalidateMode.never;
```

Reply:
Hi @skyeskie
Sorry, this PR is not ready for review yet (I probably should set it as draft).

Creating form with `autovalidateMode` to `always` `autovalidate` should not be used so it should be false. We are deprecating `autovalidate` so we are not going to use it anywhere internally. We are adding it back just because of the breaking changes!

We have an assertion that prevent devs to use both at the same time.

https://github.com/flutter/flutter/blob/f19597728f28686bfcec4a33751956aa25622fb1/packages/flutter/lib/src/widgets/form.dart#L90-L94

if you still using `autovalidate`, I suggest you to switch to `autovalidateMode`.

Reply:
Marked as WIP per comment. ;)

Reply:
The assertion works on the constructor. It doesn't prevent using `autovalidateMode` in the constructor and the `autovalidate` getter.

There are scenarios where the getter could be located in another package ([`FormBuilderCustomField`](https://github.com/danvick/flutter_form_builder/blob/v3.14.0-alpha.2/lib/src/form_builder_custom_field.dart))

Granted, I don't use that, and the `flutter_form_builder` package might even update before this change lands. It just avoids some potential issues by having a single source of truth/state.

-----

Do I need to do anything to discard the review?

Reply:
You don't need to do anything.